services/construction_service.go

```diff
@@ -365,6 +369,54 @@ func (s *ConstructionAPIService) ConstructionMetadata(
 		return nil, wrapErr(ErrGeth, err)
 	}
 
+	// Build eth transaction for L1 fee calculation
+	var toAddress common.Address
+	var ethTx *ethTypes.Transaction
+	if to != "" {
+		toAddress = common.HexToAddress(to)
+	}
+	if isEIP1559 := gasTipCap != nil && gasFeeCap != nil; isEIP1559 {
+		ethTx = ethTypes.NewTx(&ethTypes.DynamicFeeTx{
+			Nonce:     nonce,
+			GasTipCap: gasTipCap,
+			GasFeeCap: gasFeeCap,
+			Gas:       gasLimit,
+			To:        &toAddress,
```

Comment:
this should be nil, and not the zero address, if it's a contract creation.

Reply:
updated